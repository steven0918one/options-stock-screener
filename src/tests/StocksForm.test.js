import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Form from '../containers/StocksForm';
import Wrapper from './Wrapper';

Enzyme.configure({ adapter: new Adapter() });

test('Form should match snapshot', () => {
  const component = renderer.create(
    <Wrapper>
      <Form />
    </Wrapper>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Form', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Wrapper>
        <Form />
      </Wrapper>,
    );
  });
  it('is receiving and running JS', () => {
    expect(2 + 3).toEqual(5);
  });

  it('renders', () => {
    expect(wrapper).not.toBeNull();
  });

  it('starts with an empty input', () => {
    expect(wrapper.find('input').text()).toEqual('');
  });
});
