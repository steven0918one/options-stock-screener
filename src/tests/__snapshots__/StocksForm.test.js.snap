// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form i rendered 1`] = `
<form
  className="books-form"
  onSubmit={[Function]}
>
  <input
    className="title-input"
    name="title"
    onChange={[Function]}
    placeholder="Search Symbol"
    type="text"
  />
  <button
    type="submit"
  >
    <i
      className="fas fa-chart-line"
    />
  </button>
</form>
`;

exports[`Form is rendered 1`] = `
<form
  className="books-form"
  onSubmit={[Function]}
>
  <input
    className="title-input"
    name="title"
    onChange={[Function]}
    placeholder="Search Symbol"
    type="text"
  />
  <button
    type="submit"
  >
    <i
      className="fas fa-chart-line"
    />
  </button>
</form>
`;

exports[`Form should match snapshot 1`] = `
<form
  className="books-form"
  onSubmit={[Function]}
>
  <input
    className="title-input"
    name="title"
    onChange={[Function]}
    placeholder="Search Symbol"
    type="text"
  />
  <button
    type="submit"
  >
    <i
      className="fas fa-chart-line"
    />
  </button>
</form>
`;
